MIDAS/lib/ADS7138Q1/ads7138-q1.cpp

```diff
@@ -0,0 +1,60 @@
+#include "ads7138-q1.h"
+#include<Arduino.h>
+#include<Wire.h>
+
+#define WIRE Wire
+
+#define ADC_ADDR 0x14
+#define ADC_REG_READ 0x10
+#define ADC_REG_WRITE 0x08
+#define ADC_SET_BIT 0x18
+#define ADC_CLEAR_BIT 0x20
+#define ADC_REG_READ_CONTINUOUS 0x30
+#define ADC_REG_WRITE_CONTINUOUS 0x28
+#define ADC_SYSTEM_STATUS 0x00
+#define ADC_MANUAL_CH_SEL 0x11
+
+static int adc_reg_read(uint8_t address, uint8_t reg_address) {
+  WIRE.beginTransmission(address);
+  WIRE.write(ADC_REG_READ);
+  WIRE.write(reg_address);
+  if(!WIRE.endTransmission(true)){
+    Serial.println("err1");
```

Comment:
"err1", "err2", and "err3" seems a bit ambiguous.

Reply:
switched to error codes